net/src/packet.rs

```diff
@@ -248,6 +260,199 @@ impl DeParse for Packet {
                 cursor.write(transport)?;
             }
         }
+
+        match self.encap {
+            None => {
+                return Ok(NonZero::new(cursor.inner.len() - cursor.remaining)
+                    .unwrap_or_else(|| unreachable!()))
+            }
+            Some(Encap::Vxlan(ref vxlan)) => {
+                cursor.write(vxlan)?;
+            }
+        }
         Ok(NonZero::new(cursor.inner.len() - cursor.remaining).unwrap_or_else(|| unreachable!()))
     }
 }
+
+#[derive(Debug, thiserror::Error)]
+#[error("Packet already has as many VLAN headers as parser can support (max is {MAX_VLANS})")]
+pub struct TooManyVlans;
+
+impl Packet {
+    /// Create a new packet with the supplied `Eth` header.
+    fn new(eth: Eth) -> Packet {
+        Packet {
+            eth,
+            vlan: ArrayVec::default(),
+            net: None,
```

Comment:
Is there a way we can set `net` for tests in other modules? I didn't find an easy way without making the constructor for `Ipv4`, or its inner `Ipv4header`, public (e.g. `pub struct Ipv4(pub Ipv4Header);`).

Reply:
Yeah, packet generation is always tricky (especially before the whole API is reasonably stable).

In theory you should be able to use the `TypeGenerator` implementation to spit out an infinite list of `Ipv4` or `Net` objects.  That said, I need to follow up with a `TypeGenerator` for `Packet` to make this easy and sensible